dist/Data-Dumper/Dumper.xs

```diff
@@ -109,7 +109,7 @@ static I32 DD_dump (pTHX_ SV *val, const char *name, STRLEN namelen, SV *retval,
  * length parameter.  This wrongly allowed reading beyond the end of buffer
  * given malformed input */
 
-#if PERL_VERSION <= 6 /* Perl 5.6 and earlier */
+#if PERL_VERSION_LE(5,6,'*') /* Perl 5.6 and earlier */
```

Comment:
Isn't that just a weird way to write `PERL_VERSION_LE(5,7,0)`?

Reply:
this is both works, I can change it to `PERL_VERSION_LT(5,7,0)` if it makes it more clear,
I picked this syntax as I think it was a better match for the comment... but both works and are similar.

Reply:
I think the `'*'` is a very confusing and non-obvious way to write it, even if it happens to work because `char` is a numeric type in C. `PERL_VERSION_LT(5,7,0)` is _much_ clearer.

Reply:
The fact that you have to doublecheck on both the ASCII and the EBCDIC table if this correct should make it obvious that this is non-obvious

Reply:
either `PERL_VERSION_LT(5,7,0)` or `PERL_VERSION_LT(5,7)` (if it's easy to write a version that takes only two args, or has the third arg defaulting to 0) would be clearer here.

Reply:
Be careful sometimes you do not only want 0 to be the default but also need to adjust the second digit accordingly. 
Examples:
- `PERL_VERSION_LE(5.6)` should be an alias to `PERL_VERSION_LT(5,7,0)`
- `PERL_VERSION_GT(5,6)` should be similar to `PERL_VERSION_GE(5,7,0)`

Reply:
Probably a good reason that it would be simpler just to keep requiring 3 args.